build-tools-internal/src/main/groovy/elasticsearch.run-ccs.gradle

```diff
@@ -23,35 +22,25 @@ def queryingCluster = testClusters.register('querying-cluster') {
   setting 'xpack.watcher.enabled', 'false'
   setting 'xpack.ml.enabled', 'false'
   setting 'xpack.license.self_generated.type', 'trial'
-  if (proxyMode) {
-    setting 'cluster.remote.my_remote_cluster.mode', 'proxy'
-    setting 'cluster.remote.my_remote_cluster.proxy_address', {
-      "\"${fulfillingCluster.get().getAllTransportPortURI().get(0)}\""
-    }
-  } else {
-    setting 'cluster.remote.my_remote_cluster.seeds', {
-      fulfillingCluster.get().getAllTransportPortURI().collect { "\"$it\"" }.toString()
-    }
-  }
   setting 'cluster.remote.connections_per_cluster', "1"
-
   user username: 'elastic-admin', password: 'elastic-password', role: '_es_test_root'
 }
 
-// the following task is needed to make sure the fulfilling cluster is fully configured before starting both clusters
-// this allows the quering cluster to use configuration from the fulfilling cluster while honoring the RunTasks configuration (such as use port 9200)
-tasks.register('initfulfillingCluster', RunTask) {
-  useCluster testClusters.named("fulfilling-cluster")
-  initOnly = true //only initialize the testCluster, don't start it
-  portOffset = 1 //when only initializing, instruct to use one above the normal ports to avoid collisions when other cluster also initializes
-  //debug = true //this task doesn't honor the command line options for run-ccs, so need to statically configure debug
-}
 
 tasks.register("run-ccs", RunTask) {
-  dependsOn initfulfillingCluster
-  useCluster testClusters.named("fulfilling-cluster")
-  useCluster testClusters.named("querying-cluster")
+  useCluster fulfillingCluster
+  useCluster queryingCluster
   doFirst {
+    queryingCluster.get().getNodes().each { node ->
+      if (proxyMode) {
+        node.setting('cluster.remote.my_remote_cluster.mode', 'proxy')
+        node.setting('cluster.remote.my_remote_cluster.proxy_address', "\"${fulfillingCluster.get().getAllTransportPortURI().get(0)}\"")
+      } else {
+        node.setting('cluster.remote.my_remote_cluster.seeds', fulfillingCluster.get().getAllTransportPortURI().collect { "\"$it\"" }.toString())
+      }
+    }
+    queryingCluster.get().restart() //updates the on-disk config
```

Comment:
this is the main change. specifically calling .restart() before either actually start updates the configuration on disk